diff --git a/third_party/SDL/src/video/SDL_video.c b/third_party/SDL/src/video/SDL_video.c
index db2637bce1..d2d7044702 100644
--- a/third_party/SDL/src/video/SDL_video.c
+++ b/third_party/SDL/src/video/SDL_video.c
@@ -83,15 +83,15 @@ static VideoBootStrap *bootstrap[] = {
 #ifdef SDL_VIDEO_DRIVER_COCOA
     &COCOA_bootstrap,
 #endif
+#ifdef SDL_VIDEO_DRIVER_WAYLAND
+    &Wayland_bootstrap,
+#endif
 #ifdef SDL_VIDEO_DRIVER_X11
 #ifdef SDL_VIDEO_DRIVER_WAYLAND
     &Wayland_preferred_bootstrap,
 #endif
     &X11_bootstrap,
 #endif
-#ifdef SDL_VIDEO_DRIVER_WAYLAND
-    &Wayland_bootstrap,
-#endif
 #ifdef SDL_VIDEO_DRIVER_VIVANTE
     &VIVANTE_bootstrap,
 #endif
