diff --git a/Simbody/src/CablePath.cpp b/Simbody/src/CablePath.cpp
index a9a185cc..a535bbc2 100644
--- a/Simbody/src/CablePath.cpp
+++ b/Simbody/src/CablePath.cpp
@@ -55,7 +55,7 @@ PathInstanceInfo::PathInstanceInfo
         if (obs.getNumCoordsPerContactPoint() > 0) { // a surface
             mapObstacleToSurface[ox] = next++;
             mapSurfaceToObstacle.push_back(ox);
-            assert(mapSurfaceToObstacle.size() == next);
+            assert(SurfaceObstacleIndex{mapSurfaceToObstacle.size()} == next);
         }
         obstaclePose[ox]=obs.getDefaultPoseOnBody();
     }
