Comments out a line that doesn't compile with gcc

---

diff --git a/libosim/opensim-core/OpenSim/Common/Reporter.h b/libosim/opensim-core/OpenSim/Common/Reporter.h
index 9a73be1f87..01d6adc9f9 100644
--- a/libosim/opensim-core/OpenSim/Common/Reporter.h
+++ b/libosim/opensim-core/OpenSim/Common/Reporter.h
@@ -331,7 +331,7 @@ private:
             const auto& nSigFigs = chan->getOutput().getNumberOfSignificantDigits();
             // Print `value` right-justified in a column with width `_width`,
             // using `nSigFigs`: {:>{_width}.{nSigFigs}g}
-            msg += fmt::format("{:>{}.{}g}| ", value, _width, nSigFigs);
+            //msg += fmt::format("{:>{}.{}g}| ", value, _width, nSigFigs);
         }
         log_cout(msg);
 
