diff --git a/libosim/opensim-core/OpenSim/Common/IO.cpp b/libosim/opensim-core/OpenSim/Common/IO.cpp
index d3d5906a..2ea94377 100644
--- a/libosim/opensim-core/OpenSim/Common/IO.cpp
+++ b/libosim/opensim-core/OpenSim/Common/IO.cpp
@@ -31,6 +31,7 @@
 
 #include "Logger.h"
 #include <climits>
+#include <filesystem>
 #include <limits>
 #include <math.h>
 #include <string>
@@ -520,12 +521,7 @@ stod(const std::string& __str, std::size_t* __idx)
 int IO::
 makeDir(const string &aDirName)
 {
-
-#if defined __linux__ || defined __APPLE__
-    return mkdir(aDirName.c_str(),S_IRWXU);
-#else
-    return _mkdir(aDirName.c_str());
-#endif
+    return std::filesystem::create_directory(aDirName) ? 0 : 1;
 }
 //_____________________________________________________________________________
 /**
@@ -535,13 +531,9 @@ makeDir(const string &aDirName)
 int IO::
 chDir(const string &aDirName)
 {
-
-#if defined __linux__ || defined __APPLE__
-    return chdir(aDirName.c_str()); 
-#else
-    return _chdir(aDirName.c_str());
-#endif
-
+    std::error_code error;
+    std::filesystem::current_path(aDirName, error);
+    return error.value();
 }
 //_____________________________________________________________________________
 /**
@@ -551,13 +543,7 @@ chDir(const string &aDirName)
 string IO::
 getCwd()
 {
-    char buffer[PATH_MAX];
-#if defined __linux__ || defined __APPLE__
-    auto ptr = getcwd(buffer, PATH_MAX); (void)ptr;
-#else
-    _getcwd(buffer, PATH_MAX);
-#endif
-    return string(buffer);
+    return std::filesystem::current_path().string();
 }
 
 //_____________________________________________________________________________
