diff --git a/third_party/libosim/opensim-core/OpenSim/Simulation/Model/Model.cpp b/third_party/libosim/opensim-core/OpenSim/Simulation/Model/Model.cpp
index 7ff23bc6b..4cb6392aa 100644
--- a/third_party/libosim/opensim-core/OpenSim/Simulation/Model/Model.cpp
+++ b/third_party/libosim/opensim-core/OpenSim/Simulation/Model/Model.cpp
@@ -798,11 +798,13 @@ void Model::extendFinalizeFromProperties()
 {
     Super::extendFinalizeFromProperties();
 
-    // wipe-out the existing System
+    // wipe-out the existing System: make sure to wipe subsystems first
     _matter.reset();
     _forceSubsystem.reset();
     _contactSubsystem.reset();
-    _system.reset();
+    _cableSubsystem.reset();
+
+    _system.reset();  // after wiping all subsystems
 
     if(getForceSet().getSize()>0)
     {
diff --git a/third_party/libosim/opensim-core/OpenSim/Simulation/Model/Model.h b/third_party/libosim/opensim-core/OpenSim/Simulation/Model/Model.h
index 9f2741db1..30e001123 100644
--- a/third_party/libosim/opensim-core/OpenSim/Simulation/Model/Model.h
+++ b/third_party/libosim/opensim-core/OpenSim/Simulation/Model/Model.h
@@ -1154,6 +1154,7 @@ public:
         _forceSubsystem.reset();
         _gravityForce.reset();
         _matter.reset();
+        _cableSubsystem.reset();
     }
 
     /** Override of the default implementation to account for versioning. */
