diff --git a/OpenSim/Simulation/Model/GeometryPath.cpp b/OpenSim/Simulation/Model/GeometryPath.cpp
index 4480ba2e7..d507f6596 100644
--- a/OpenSim/Simulation/Model/GeometryPath.cpp
+++ b/OpenSim/Simulation/Model/GeometryPath.cpp
@@ -376,61 +376,26 @@ void GeometryPath::
 getPointForceDirections(const SimTK::State& s, 
                         OpenSim::Array<PointForceDirection*> *rPFDs) const
 {
-    int i;
-    AbstractPathPoint* start;
-    AbstractPathPoint* end;
-    const OpenSim::PhysicalFrame* startBody;
-    const OpenSim::PhysicalFrame* endBody;
-    const Array<AbstractPathPoint*>& currentPath = getCurrentPath(s);
-
-    int np = currentPath.getSize();
-    rPFDs->ensureCapacity(np);
-    
-    for (i = 0; i < np; i++) {
-        PointForceDirection *pfd = 
-            new PointForceDirection(currentPath[i]->getLocation(s), 
-                                    currentPath[i]->getParentFrame(), Vec3(0));
-        rPFDs->append(pfd);
-    }
-
-    for (i = 0; i < np-1; i++) {
-        start = currentPath[i];
-        end = currentPath[i+1];
-        startBody = &start->getParentFrame();
-        endBody = &end->getParentFrame();
-
-        if (startBody != endBody)
+    class PFDConsumer final : public ForceConsumer {
+    public:
+        PFDConsumer(OpenSim::Array<PointForceDirection*>* pfds) :
+            m_PFDs{pfds}
+        {}
+    private:
+        void implConsumePointForce(
+            const SimTK::State& state,
+            const PhysicalFrame& frame,
+            const SimTK::Vec3& point,
+            const SimTK::Vec3& force)
         {
-            Vec3 posStart, posEnd;
-            Vec3 direction(0);
-
-            // Find the positions of start and end in the inertial frame.
-            //engine.getPosition(s, start->getParentFrame(), start->getLocation(), posStart);
-            posStart = start->getLocationInGround(s);
-            
-            //engine.getPosition(s, end->getParentFrame(), end->getLocation(), posEnd);
-            posEnd = end->getLocationInGround(s);
-
-            // Form a vector from start to end, in the inertial frame.
-            direction = (posEnd - posStart);
-
-            // Check that the two points are not coincident.
-            // This can happen due to infeasible wrapping of the path,
-            // when the origin or insertion enters the wrapping surface.
-            // This is a temporary fix, since the wrap algorithm should
-            // return NaN for the points and/or throw an Exception- aseth
-            if (direction.norm() < SimTK::SignificantReal){
-                direction = direction*SimTK::NaN;
-            }
-            else{
-                direction = direction.normalize();
-            }
-
-            // Get resultant direction at each point 
-            rPFDs->get(i)->addToDirection(direction);
-            rPFDs->get(i+1)->addToDirection(-direction);
+            m_PFDs->append(new PointForceDirection{point, frame, force});
         }
-    }
+
+        OpenSim::Array<PointForceDirection*>* m_PFDs;
+    };
+
+    PFDConsumer consumer{rPFDs};
+    produceForces(s, 1.0, consumer);
 }
 
 void GeometryPath::produceForces(const SimTK::State& s,
@@ -484,10 +449,20 @@ void GeometryPath::produceForces(const SimTK::State& s,
             previousDirection = SimTK::Vec3(0.0f);
         }
 
-        // If applicable, produce the force
-        if (force != SimTK::Vec3(0.0)) {
-            forceConsumer.consumePointForce(s, currentPoint.getParentFrame(), currentPoint.getLocation(s), force);
-        }
+        // A point force should be emitted even if it's zero to ensure that the legacy
+        // behavior of `getPointForceDirections` is maintained.
+        //
+        // `getPointForceDirections` always produces one point force per path
+        // point, even if the force is empty, and legacy codebases use `getPointForceDirections`
+        // to figure out the difference between anatomical/effective lines of action:
+        //
+        //     https://github.com/modenaxe/MuscleForceDirection/blob/60d0e39b918626494e8b79f93d1a5eba9054b62e/CPP/MuscleForceDirection/MuscleForceDirection.cpp
+        forceConsumer.consumePointForce(
+            s,
+            currentPoint.getParentFrame(),
+            currentPoint.getLocation(s),
+            force
+        );
     }
 }
 
diff --git a/OpenSim/Simulation/Model/Ligament.cpp b/OpenSim/Simulation/Model/Ligament.cpp
index b05cd5c54..151a3c45e 100644
--- a/OpenSim/Simulation/Model/Ligament.cpp
+++ b/OpenSim/Simulation/Model/Ligament.cpp
@@ -25,7 +25,6 @@
 // INCLUDES
 //=============================================================================
 #include "Ligament.h"
-#include "PointForceDirection.h"
 
 #include <OpenSim/Common/Assertion.h>
 #include <OpenSim/Common/SimmSpline.h>
